File: Java/Easy/Accepted/substrings-of-size-three-with-distinct-characters_3.java
class Solution {
    public int countGoodSubstrings(String s) {
        
        if (s.length() < 3) {
            return 0;
        }
        int count = 0;
        for (int i = 0; i <= s.length() - 3; i++) {
            if (s.charAt(i) != s.charAt(i + 1) && s.charAt(i) != s.charAt(i + 2) && s.charAt(i + 1) != s.charAt(i + 2)) {
                count++;
            }
        }
        return count;
    }
}

File: Rust/Easy/RuntimeError/substrings-of-size-three-with-distinct-characters_3.rs
impl Solution {
    pub fn count_good_substrings(s: String) -> i32 {
        
        let mut res = 0;
        let chars: Vec<char> = s.chars().collect();
        for i in 0..(chars.len() - 2) {
            if chars[i] != chars[i + 1] && chars[i + 1] != chars[i + 2] && chars[i] != chars[i + 2] {
                res += 1;
            }
        }
        res
    }
}